arcgis-ios-sdk-samples/Maps/Show location history/LocationHistoryViewController.swift

```diff
@@ -0,0 +1,191 @@
+//
+// Copyright © 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+// MARK: - Constants
+
+private enum Constants {
+    static let locationBorderColor = UIColor.backgroundGray
+    static let locationInnerColor = UIColor.primaryTextColor
+    static let trackColor = UIColor.primaryBlue.withAlphaComponent(0.5)
+
+    static let locationDiameter: CGFloat = 14
+    static let locationBorderWidth: CGFloat = 2
+    static let trackWidth: CGFloat = 10
+
+    static let initialZoomScale: Double = 10000
+}
+
+// MARK: - LocationTrackingStatus
+
+private enum LocationTrackingStatus {
+    case paused
+    case inProgress
+}
+
+// MARK: - LocationHistoryViewController
+
+class LocationHistoryViewController: UIViewController {
+    @IBOutlet private weak var mapView: AGSMapView!
+    @IBOutlet private weak var trackingBarButtonItem: UIBarButtonItem!
+    
+    private var currentStatus: LocationTrackingStatus = .paused {
+        didSet {
+            handleLocationStatusChange()
+        }
+    }
+    
+    private let map = AGSMap(basemap: .lightGrayCanvasVector())
```

Comment:
This variable is only being used in `setupMapView()`, so I would suggest just moving it there. The map view holds a strong reference to the map, so we don't need to hold onto it ourselves unnecessarily.

Reply:
Addressed in [`01290cd0`](https://github.com/Esri/arcgis-runtime-samples-ios/pull/751/commits/01290cd0bd89320c0cf3d012ebce664890fb31dd).